src/coin/trx/transactionBuilder.ts

```
@@ -113,9 +114,13 @@ export class TransactionBuilder extends BaseTransactionBuilder {
     }
   }
 
-  validateKey(key: BaseKey) {
-    // TODO: determine valid key format
-    return true;
+  validateKey(key: BaseKey): boolean {
```

JSDocs

---

Added. I thought JSDoc for a self-explainable function is not mandatory. Also added JSDoc plugin for Eslint.
After fixing linter issues across `transactionBuilder.ts` I noticed that all the `validateSomething` functions do not return any value thus the name `isKeyValid` makes more sense for `validateKey`. Just a suggestion.

---

Thanks for the plugin, that is great!

You are absolutely right. In this case, the `return true` placeholder was misleading. This method should throw if the key is invalid somehow, not return a boolean.

It is implicitly used here in `baseTransactionBuilder.ts`:
```
  sign(key: BaseKey): void {
    // Make sure the key is valid
    this.validateKey(key);
    if (!this.transaction.canSign(key)) {
      throw new SigningError('Private key cannot sign the transaction');
    }
    this.transaction = this.signImplementation(key);
  }
```

---

Updated